### ISSUE TITLE
Field should execute every validator

### ISSUE THREAD

**Issue by tflori** [tflori/verja#19]
When you create a field with a filter with the same name as the validator, it will add only a filter. When you then validate this field it does not execute the validator (there is only a filter).

```
>>> $field = new Verja\Field(['integer']);
=> Verja\Field {#208}
>>> $field->validate('foo');
=> true
```

What we need to do is to filter the value before we execute the validators but then we filter twice - once from Gate and once from validator (with the already filtered value - so the cache does not hit).

**Comment by tflori** [comment]
The root of the problem is that `Gate` users `Field::filter()` and the filtered value for `Field::validate()`. This logic has to be moved to `Field::validate()`. Then we can call `Field::validate()` directly when we want to filter and validate and `Field::filter()` when we only want to filter.

```php
class Gate {
  public function validate($data) {
    foreach ($this->field as $key => $field) {
      $field->validate($data[$key], $data);
    }
  }
}

class Field {
  public function filter($value, $context = []) {
    $filtered = $value; // what ever
    return $filtered;
  }
  public function validate($value, $context = []) {
    $filtered = $this->filter($value, $context);
    $valid = false; // what ever
    return $valid;
  }
}
```

**Comment by tflori** [comment]
solved with #21
